File: dice.cpp
#include "dice.hpp"

Die::Die(std::vector<DieSide> setupSides){
    for (auto side :setupSides){
        sides.push_back(side);
    }
}
DieSide* Die::getSide(int idx){
    return &sides[idx];
}
std::vector<DieSide> Die::getSides(){
    return sides;
}
void Die::setSide(int idx,DieSide side){
    sides[idx] = side;
}
void Die::roll(){
    lastRoll = sides[GetRandomValue(0,5)].getNumber();
}
int Die::getLastRoll(){
    return lastRoll;
}

DieSide::DieSide(int initalNumber){
    number = initalNumber;
}
int DieSide::getNumber(){
    return number;
}
void DieSide::setNumber(int newNumber){
    number = newNumber;
}
void DieSide::addToNumber(int addingValue){
    number += addingValue;
}

File: dice.py
import random

class Die_face:
    def __init__(self,value:int) -> None:
        self.value = value
        
class Die:
    def __init__(self,faces:list[Die_face]) -> None:
        self.faces = faces
        self.last_roll = 1
    def roll(self) -> int:
        roll = random.choice(self.faces).value
        self.last_roll = roll
        return roll
    
STANDARD_DIE = Die([Die_face(1),Die_face(2),Die_face(3),Die_face(4),Die_face(5),Die_face(6)])